pallets/schemas/src/lib.rs

```diff
@@ -120,6 +124,15 @@ pub mod pallet {
 			Ok(cur_count)
 		}
 
+		pub fn get_schema_by_id(
+			schema_id: SchemaId,
+		) -> Result<Option<SchemaResponse>, DispatchError> {
+			if let Some(schema) = Self::get_schema(schema_id) {
+				return Ok(Some(schema.map_to_response(schema_id)))
```

Comment:
It not very clear to me on how we could return DispatchError?
maybe we just need `<Option<SchemaResponse>`

Reply:
So I tried returning `<Option<SchemaResponse>` but my rpc was giving an unrelated error. I tried for a while but only changing the return type from `<Option<SchemaResponse>` to `Result<Option<SchemaResponse>, DispatchError>` solved it.

I think we might be able to come back and change this later if we have some extra time.

Reply:
does map_to_response return any error?

Reply:
returning `Option`

Reply:
Outdated comment